build/.azure-pipelines.TemplateValidation.yml

```diff
@@ -4,79 +4,209 @@ jobs:
   pool:
     vmImage: 'windows-2022'
 
+  # We run two validations per CI machine.
+  # When adding a new validation:
+  # - If it makes the number of validations odds:
+  #    - Add the new validation to templateArgs1, and use the value PLACEHOLDER for templateArgs2
+  #    - Note that the value PLACEHOLDER is special cased to not actually run.
+  # - If it makes the number of validations even:
+  #    - There should already be an existing templateArgs2 with value PLACEHOLDER
+  #    - So, replace the PLACEHOLDER with the arguments to test.
+  # Notes:
+  # 1. We name the matrix elements as "TestCase1_And_TestCase2"
+  # 2. Whenever we have an odd number of cases and are using PLACEHOLDER for the second, you can name it as "TestCase1",
+  #    But whenever the PLACEHOLDER is replaced with a new test case, don't forget to rename it to "TestCase1_And_TestCase2"
+  # 3. It's always best to have a good grouping of test cases to reduce the work done in CI. For example,
+  #    always try to group net7 test cases together, and net8 test cases together (same goes for maui and non-maui), etc..
+  #    This is to reduce the amount of restore work needs to be done in a single job (running on a single CI machine).
   strategy:
     matrix:
-      DefaultArguments:
-        templateArgs: ''
-      Blank:
-        templateArgs: '-preset blank'
-      SkiaOnlyHeads:
-        templateArgs: '-platforms gtk wpf linux-fb'
-      MobileOnlyHeads:
-        templateArgs: '-platforms android ios maccatalyst'
-      BlankMarkup:
-        templateArgs: '-preset blank -markup csharp'
-      BlankNoCpm:
-        templateArgs: '-preset blank --cpm false'
-      MVVM:
-        templateArgs: '-presentation mvvm'
-      NoCentralPackageManagement:
-        templateArgs: '--cpm false'
-      CSharpMarkup:
-        templateArgs: '-markup csharp'
-      NoHosting:
-        templateArgs: '-di false'
-      HostingOnly:
-        templateArgs: '-config false -loc false -http false -log none --navigation blank'
-      NoConfiguration:
-        templateArgs: '-config false'
-      NoLocalization:
-        templateArgs: '-loc false'
-      NoHttp:
-        templateArgs: '-http false'
-      NoSerilog:
-        templateArgs: '-log default'
-      NoServer:
-        templateArgs: '-server false'
-      NoServerNoHttp:
-        templateArgs: '-server false -http false'
-      NoTests:
-        templateArgs: '-tests none'
-      FrameNavigation:
-        templateArgs: '--navigation blank'
-      Net8:
-        templateArgs: '-tfm net8.0'
+      DefaultArguments_And_Blank:
+        templateArgs1: ''
+        templateArgs2: '-preset blank'
+        net7_workloads: 'android ios maccatalyst'
+        net8_workloads: ''
+        install_net7: true
+        install_net8: false
```

Comment:
I'm not sure why these can't be calculated based on templateArgs1 and templateArgs2

Reply:
For workloads, it requires a bit more work, for example, if we ever had a complete job with Skia-only validations, we will want to skip workloads. So it needs to analyze the platforms flags properly.

Reply:
This parameter may be useful: https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/jobs-job-strategy?view=azure-pipelines#strategy-matrix-maxparallel

Reply:
Right now the idea is to make two validations on a single agent, rather than just limiting the number of parallel agents.

Reply:
@dansiegel are you ok with these additional parameters, or would you prefer to see them calculated?